src/filepath.c

```diff
@@ -1424,7 +1424,7 @@ f_readdir(typval_T *argvars, typval_T *rettv)
     expr = &argvars[1];
 
     ret = readdir_core(&ga, path, FALSE, (void *)expr,
-	    (expr->v_type == VAR_UNKNOWN) ? NULL : readdir_checkitem);
+	    (expr->v_type == VAR_UNKNOWN) ? NULL : readdir_checkitem, 0);
```

Comment:
This changes the current behavior. Should be:
```suggestion
	    (expr->v_type == VAR_UNKNOWN) ? NULL : readdir_checkitem, READDIR_SORT_BYTE);
```

Reply:
yes makes sense, was an overview.